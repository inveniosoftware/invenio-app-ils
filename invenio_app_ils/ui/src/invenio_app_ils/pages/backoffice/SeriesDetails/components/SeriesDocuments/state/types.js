export const IS_LOADING = 'fetchSeriesDocuments/IS_LOADING';
export const SUCCESS = 'fetchSeriesDocuments/SUCCESS';
export const HAS_ERROR = 'fetchSeriesDocuments/HAS_ERROR';
