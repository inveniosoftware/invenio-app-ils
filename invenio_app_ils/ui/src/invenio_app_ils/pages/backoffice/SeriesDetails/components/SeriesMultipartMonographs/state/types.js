export const IS_LOADING = 'fetchSeriesMultipartMonographs/IS_LOADING';
export const SUCCESS = 'fetchSeriesMultipartMonographs/SUCCESS';
export const HAS_ERROR = 'fetchSeriesMultipartMonographs/HAS_ERROR';
