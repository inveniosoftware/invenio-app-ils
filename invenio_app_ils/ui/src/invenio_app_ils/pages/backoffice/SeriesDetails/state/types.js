export const IS_LOADING = 'fetchSeriesDetails/IS_LOADING';
export const SUCCESS = 'fetchSeriesDetails/SUCCESS';
export const HAS_ERROR = 'fetchSeriesDetails/HAS_ERROR';
