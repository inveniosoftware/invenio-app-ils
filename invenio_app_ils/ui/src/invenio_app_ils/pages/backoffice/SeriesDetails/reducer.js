export { default as seriesDetailsReducer } from './state/reducer';
export {
  default as seriesDocumentsReducer,
} from './components/SeriesDocuments/state/reducer';
