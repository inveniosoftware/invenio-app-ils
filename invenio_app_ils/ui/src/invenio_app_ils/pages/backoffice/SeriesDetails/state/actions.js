import {
  DELETE_IS_LOADING,
  DELETE_SUCCESS,
  DELETE_HAS_ERROR,
  IS_LOADING,
  SUCCESS,
  HAS_ERROR,
} from './types';
import { ES_DELAY } from '../../../../common/config';
import { goTo } from '../../../../history';
import { series as seriesApi } from '../../../../common/api';
import { BackOfficeRoutes } from '../../../../routes/urls';
import {
  sendErrorNotification,
  sendSuccessNotification,
} from '../../../../common/components/Notifications';

export const deleteSeries = seriesPid => {
  return async dispatch => {
    dispatch({
      type: DELETE_IS_LOADING,
    });

    try {
      await seriesApi.delete(seriesPid);
      dispatch({
        type: DELETE_SUCCESS,
        payload: { seriesPid: seriesPid },
      });
      dispatch(
        sendSuccessNotification(
          'Success!',
          `The series ${seriesPid} has been deleted.`
        )
      );
      setTimeout(() => {
        goTo(BackOfficeRoutes.seriesList);
      }, ES_DELAY);
    } catch (error) {
      dispatch({
        type: DELETE_HAS_ERROR,
        payload: error,
      });
      dispatch(sendErrorNotification(error));
    }
  };
};

export const fetchSeriesDetails = seriesPid => {
  return async dispatch => {
    dispatch({
      type: IS_LOADING,
    });

    await seriesApi
      .get(seriesPid)
      .then(response => {
        dispatch({
          type: SUCCESS,
          payload: response.data,
        });
      })
      .catch(error => {
        dispatch({
          type: HAS_ERROR,
          payload: error,
        });
        dispatch(sendErrorNotification(error));
      });
  };
};

export const updateSeries = (seriesPid, path, value) => {
  return async dispatch => {
    dispatch({
      type: IS_LOADING,
    });

    await seriesApi
      .patch(seriesPid, [
        {
          op: 'replace',
          path: path,
          value: value,
        },
      ])
      .then(response => {
        dispatch({
          type: SUCCESS,
          payload: response.data,
        });
        dispatch(
          sendSuccessNotification(
            'Success!',
            `The series ${seriesPid} has been updated.`
          )
        );
      })
      .catch(error => {
        dispatch({
          type: HAS_ERROR,
          payload: error,
        });
        dispatch(sendErrorNotification(error));
      });
  };
};
