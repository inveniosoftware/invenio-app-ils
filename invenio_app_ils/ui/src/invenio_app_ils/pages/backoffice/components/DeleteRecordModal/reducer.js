export { default as deleteRecordModalReducer } from './state/reducer';
