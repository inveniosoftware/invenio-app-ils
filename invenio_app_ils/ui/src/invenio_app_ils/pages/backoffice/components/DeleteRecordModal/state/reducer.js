import { IS_LOADING, SUCCESS, HAS_ERROR } from './types';

export const initialState = {
  data: [],
  error: {},
  hasError: false,
  isLoading: false,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case IS_LOADING:
      return { ...state, isLoading: true };
    case SUCCESS:
      return {
        ...state,
        isLoading: false,
        data: action.payload,
        error: {},
        hasError: false,
      };
    case HAS_ERROR:
      return {
        ...state,
        isLoading: false,
        data: [],
        error: action.payload,
        hasError: true,
      };
    default:
      return state;
  }
};
