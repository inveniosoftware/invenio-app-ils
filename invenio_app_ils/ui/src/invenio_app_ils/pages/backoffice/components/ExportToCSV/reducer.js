export { default as csvExportReducer } from './state/reducer';
