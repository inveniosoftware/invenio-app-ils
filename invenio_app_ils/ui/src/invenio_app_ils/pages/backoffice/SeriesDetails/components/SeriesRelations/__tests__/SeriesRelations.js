import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import { SeriesRelations } from '../';
import { Settings } from 'luxon';
import { initialState } from '../state/reducer';
import { initialState as seriesState } from '../../../state/reducer';
import history from '../../../../../../history';
import { BackOfficeRoutes } from '../../../../../../routes/urls';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

Settings.defaultZoneName = 'utc';
jest.mock('../../../../../../common/config/invenioConfig');

describe('Series relations tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  let store;
  beforeEach(() => {
    store = mockStore({
      seriesDetails: seriesState,
      seriesRelations: initialState,
    });
    store.clearActions();
  });

  it('should load the relations component', () => {
    const relations = {};
    const component = shallow(<SeriesRelations relations={relations} />);
    expect(component).toMatchSnapshot();
  });

  it('should render tabs', () => {
    seriesState.isLoading = false;
    seriesState.data = {
      id: '1',
      metadata: {
        pid: '1',
        mode_of_issuance: 'MULTIPART_MONOGRAPH',
        relations: {},
      },
    };
    const state = initialState;
    state.isLoading = false;
    store = mockStore({
      seriesDetails: seriesState,
      seriesRelations: state,
      esSelector: { selections: [] },
    });

    component = mount(
      <Provider store={store}>
        <SeriesRelations />
      </Provider>
    );

    const buttons = component.find('ManageRelationsButton');
    expect(buttons).toHaveLength(2);
  });

  it('should render pagination for many rows', () => {
    const state = initialState;
    state.data = {
      edition: [
        {
          pid: '2',
          pid_type: 'docid',
        },
        {
          pid: '3',
          pid_type: 'docid',
        },
      ],
    };
    state.isLoading = false;
    seriesState.isLoading = false;
    seriesState.data = {
      id: '1',
      metadata: {
        pid: '1',
        mode_of_issuance: 'MULTIPART_MONOGRAPH',
        relations: state.data,
      },
    };
    store = mockStore({
      seriesDetails: seriesState,
      seriesRelations: state,
      esSelector: { selections: [] },
    });

    component = mount(
      <Provider store={store}>
        <SeriesRelations showMaxRows={1} />
      </Provider>
    );

    const editionTable = component
      .find('ResultsTable')
      .filterWhere(el => el.prop('name') === 'related editions');
    expect(editionTable.contains('Pagination'));
    const pagination = editionTable.find('Pagination').first();
    expect(pagination.prop('currentSize')).toEqual(1);
    expect(pagination.prop('totalResults')).toEqual(2);
  });

  it('should go to record details when clicking on a row', () => {
    const mockedHistoryPush = jest.fn();
    history.push = mockedHistoryPush;

    const state = initialState;
    const id = '2';
    state.data = {
      edition: [
        {
          pid: id,
          pid_type: 'docid',
        },
        {
          pid: '3',
          pid_type: 'docid',
        },
      ],
    };
    state.isLoading = false;
    seriesState.isLoading = false;
    seriesState.data = {
      id: '1',
      metadata: {
        pid: '1',
        mode_of_issuance: 'MULTIPART_MONOGRAPH',
        relations: state.data,
      },
    };
    store = mockStore({
      seriesDetails: seriesState,
      seriesRelations: state,
      esSelector: { selections: [] },
    });

    component = mount(
      <Provider store={store}>
        <SeriesRelations showMaxRows={1} />
      </Provider>
    );

    const button = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === id)
      .find('i');
    button.simulate('click');
    const expectedUrl = BackOfficeRoutes.documentDetailsFor(id);
    const expectedState = { pid: id, type: 'Document' };
    expect(mockedHistoryPush).toHaveBeenCalledWith(expectedUrl, expectedState);
  });
});
