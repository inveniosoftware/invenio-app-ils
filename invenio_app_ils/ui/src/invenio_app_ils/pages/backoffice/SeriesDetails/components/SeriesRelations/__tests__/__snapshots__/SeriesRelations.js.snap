// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Series relations tests should load the relations component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
